spec/create_new_milestone_action_spec.rb

```diff
@@ -0,0 +1,160 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::CreateNewMilestoneAction do
+  let(:test_repository) { 'test-repository' }
+  let(:test_token) { 'Test-GithubToken-1234' }
+  let(:test_milestone) do
+    { title: '10.1', number: '1234', due_on: '2022-10-31T07:00:00Z' }
+  end
+  let(:milestone_list) do
+    [
+      { title: '10.2', number: '1234', due_on: '2022-10-31T12:00:00Z' },
+      { title: '10.3', number: '4567', due_on: '2022-11-02T15:00:00Z' },
+      { title: '10.4', number: '7890', due_on: '2022-11-04T23:59:00Z' },
+    ]
+  end
+  let(:client) do
+    instance_double(
+      Octokit::Client,
+      list_milestones: [test_milestone],
+      create_milestone: nil,
+      user: instance_double('User', name: 'test'),
+      'auto_paginate=': nil
+    )
+  end
+
+  before do
+    allow(Octokit::Client).to receive(:new).and_return(client)
+  end
+
+  it 'computes the correct due date and milestone description' do
+    comment = "Code freeze: November 14, 2022\nApp Store submission: November 28, 2022\nRelease: November 28, 2022\n"
+    expect(client).to receive(:create_milestone).with(test_repository, '10.2', due_on: '2022-11-14T12:00:00Z', description: comment)
+    run_described_fastlane_action(default_params)
+  end
+
+  it 'removes 3 days from the AppStore submission date when `:need_appstore_submission` is true' do
+    comment = "Code freeze: November 14, 2022\nApp Store submission: November 25, 2022\nRelease: November 28, 2022\n"
+    expect(client).to receive(:create_milestone).with(test_repository, '10.2', due_on: '2022-11-14T12:00:00Z', description: comment)
+    run_action_with(:need_appstore_submission, true)
+  end
+
+  it 'uses the most recent milestone date to calculate the due date and version of new milestone' do
+    comment = "Code freeze: November 18, 2022\nApp Store submission: December 02, 2022\nRelease: December 02, 2022\n"
+    allow(client).to receive(:list_milestones).and_return(milestone_list)
+    expect(client).to receive(:create_milestone).with(test_repository, '10.5', due_on: '2022-11-18T12:00:00Z', description: comment)
+    run_described_fastlane_action(default_params)
+  end
+
+  it 'raises an error when the due date of milestone does not exists' do
+    allow(client).to receive(:list_milestones).and_return([{ title: '10.1', number: '1234' }])
+    expect { run_described_fastlane_action(default_params) }.to raise_error(FastlaneCore::Interface::FastlaneError, 'Milestone 10.1 has no due date.')
+  end
+
+  it 'raises an error when the milestone is not found or does not exist on the repository' do
+    allow(client).to receive(:list_milestones).and_return([])
+    expect { run_described_fastlane_action(default_params) }.to raise_error(FastlaneCore::Interface::FastlaneError, 'No milestone found on the repository.')
+  end
```

Comment:
Nice :+1:

nit: how about grouping those in `describe 'date computation is correct'` (for the first 3) and `describe 'raises error when it can't find last milestone'` (or maybe `context 'when last milestone cannot be used'`?) (for the last 2) for example?

Reply:
Yeah! It will be more organized 😄